app/lib/submission_builder/authentication_header.rb

```diff
@@ -42,7 +42,7 @@ def document
               xml.IPv6AddressTxt device_info&.ip_address if device_info&.ip_address&.ipv6?
             end
             xml.IPTs datetime_type(device_info&.updated_at)
-            xml.DeviceId device_info&.device_id || 'AB' * 20
+            xml.DeviceId /^[A-F0-9]{40}$/.match?(device_info&.device_id) ? device_info&.device_id : 'AB' * 20
```

Comment:
can we write a test for this?

Reply:
@anisharamnani done! thanks for adding that, it actually helped me catch that I hadn't added it to the initial device info block so glad you pointed that out